src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -284,13 +283,12 @@ public byte[] ToByteArray()
             byte highByte;
             int nonZeroDwordIndex = 0;
             uint highDword;
-            uint[] bits = _bits;
-            if (bits == null)
+            if (_bits == null)
             {
-                highByte = (byte)((sign < 0) ? 0xff : 0x00);
-                highDword = unchecked((uint)sign);
+                highByte = (byte)((_sign < 0) ? 0xff : 0x00);
+                highDword = unchecked((uint)_sign);
```

Comment:
So the problem with loading SOS was resolved by running the 32-bit version of WinDbg instead of the 64-bit version.

It appears that whether the members are `readonly` does not affect the JITted assembly.

I have created a Gist with x86 assembly dumps for three variants:
  * "using_locals" - uses local vars to store `_sign` and `_bits` (the version merged in PR #3509).
  * "access_readonly_members" - accesses `_sign` and `_bits` directly, which are marked `readonly` (the version of ToByteArray() in commit 7465c58)
  * "access_members" - accesses `_sign` and `_bits` directly, but these fields are **not** marked `readonly`

https://gist.github.com/dtrebbien/7e4b8496b7dd6b9bc4b2

Other than the code addresses, there does not appear to be a difference between the assembly generated for the "access_readonly_members" and "access_members" variants.  However, the generated assembly for the "using_locals" variant is simpler.

For example, consider the `if (_bits == null)` check in:

            bool needExtraByte = (msb & 0x80) != (highByte & 0x80);
            byte[] bytes;
            int curByte = 0;
            if (_bits == null)

The assembly generated for the "using_locals" variant is:
<pre>
02e90710 837dcc00        cmp     dword ptr [ebp-34h],0
02e90714 7533            jne     02e90749
</pre>

The assembly generated for the "access_readonly_members" variant is:
<pre>
011d0728 8b45d0          mov     eax,dword ptr [ebp-30h]
011d072b 833800          cmp     dword ptr [eax],0
011d072e 7535            jne     011d0765
</pre>